controllers/controlplane_controller.go

```diff
@@ -127,12 +170,12 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	debug(log, "creating MTLS certificate", controlplane)
-	created, certSecretName, err := r.ensureCertificate(ctx, controlplane)
+	created, certSecret, err := r.ensureCertificate(ctx, controlplane)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	if created {
-		return ctrl.Result{Requeue: true, RequeueAfter: requeueWithoutBackoff}, nil // TODO: remove after https://github.com/Kong/gateway-operator/issues/26
+		return ctrl.Result{}, nil
```

Comment:
might be worth a breadcrumb comment here for future readers:

```suggestion
		return ctrl.Result{}, nil // requeue will be triggered by the creation of the owned object
```

There's actually several places where something similar happens, I know it might seem excessive but I would personally just put a comment like this on each of them.